pysodium/__init__.py

```diff
@@ -153,6 +157,33 @@ def crypto_aead_chacha20poly1305_decrypt(ciphertext, ad, nonce, key):
     __check(sodium.crypto_aead_chacha20poly1305_decrypt(m, ctypes.byref(mlen), None, ciphertext, clen, ad, adlen, nonce, key))
     return m.raw
 
+# crypto_aead_chacha20poly1305_ietf_encrypt(unsigned char *c, unsigned long long *clen_p, const unsigned char *m, unsigned long long mlen, const unsigned char *ad, unsigned long long adlen, const unsigned char *nsec, const unsigned char *npub, const unsigned char *k)
+def crypto_aead_chacha20poly1305_ietf_encrypt(message, ad, nonce, key):
+
+    if ad == None: raise ValueError("ad cannot be None")
+    mlen = ctypes.c_ulonglong(len(message))
+
+    if ad:
```

Comment:
this and the if in 161 is unnecessary there will be an exception anyway on len() i removed this and also in other functions in my branch